support-frontend/assets/components/subscriptionCheckouts/address/__tests__/deliveryCheckTests.js

```diff
@@ -0,0 +1,13 @@
+import { postcodeIsWithinDeliveryArea } from 'components/subscriptionCheckouts/address/deliveryCheck';
+
+describe('Delivery Check', () => {
+  it('should return true if the postcode is in the delivery area', () => {
+    expect(postcodeIsWithinDeliveryArea('SE2 2LB')).toBeTruthy();
+    expect(postcodeIsWithinDeliveryArea('SE20 2LB')).toBeTruthy();
+    expect(postcodeIsWithinDeliveryArea('SE8 2AD')).toBeTruthy();
+    expect(postcodeIsWithinDeliveryArea('SW17 2LB')).toBeTruthy();
```

Comment:
the first 4 tests are pretty much the same logic, why not just reduce to one.
Also add some partial and empty string, if that's something the code might have to deal with? e.g.
"", "N", "N1", "N111111111"

Reply:
I tend to agree - it would be great to break out all the steps involved in manipulating the postcode and test each one rather than multiple tests of the overall outcome.

Reply:
this only really has two tests 'is the prefix inside the M25' or 'is the prefix outside the M25' all the other validation happens when we check if this is a valid postcode, before this M25 check

Reply:
fair enough in which case the first 4 tests are all the same by the looks in terms of logic?  Although more tests is always stated as a goal, it's actually better to avoid redundant tests as they have a maintenance cost.